src/messages/search/MessageFlagsPredicate.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "common/FlagsEnum.hpp"
+#include "messages/search/MessagePredicate.hpp"
+
+namespace chatterino {
+
+using MessageFlags = FlagsEnum<MessageFlag>;
+
+/**
+ * @brief MessagePredicate checking for message flags.
+ *
+ * This predicate will only allow messages with a list of flags.
+ * specified by user friendly names for the flags.
+ */
+class MessageFlagsPredicate : public MessagePredicate
+{
+public:
+    /**
+     * @brief Create a MessageFlagsPredicate with a list of flags to search for.
+     *
+     * @param flags a list of names for the flags a message should have
+     */
```

Comment:
I propose adding the user-friendly names allowed (`deleted`, `sub`, `highlighted` etc.) to the documentation here